core/citrus-spring/src/main/java/com/consol/citrus/common/XmlTestLoader.java

```diff
@@ -91,7 +98,25 @@ private ApplicationContext loadApplicationContext() {
                     new String[]{
                             getSource(),
                             "com/consol/citrus/spring/annotation-config-ctx.xml"},
-                    true, getParentApplicationContext());
+                    true, getParentApplicationContext()) {
+
+                /**
+                 * Configures the namespace handler resolver with the {@link XmlTestLoaderConfiguration}
+                 * @param reader
+                 */
+                @Override
+                protected void initBeanDefinitionReader(XmlBeanDefinitionReader reader) {
+
+                    XmlTestLoaderConfiguration loaderConfiguration = testClass.getAnnotation(XmlTestLoaderConfiguration.class);
+                    if (loaderConfiguration != null) {
+                        reader.setNamespaceHandlerResolver(new TestLoaderConfigurationApplicator(loaderConfiguration, reader.getNamespaceHandlerResolver()));
+                    }
+                    super.initBeanDefinitionReader(reader);
```

Comment:
I think `super`-invocation should stand at the beginning, no? because it would possibly overwrite `reader#setNamespaceHandlerResolver` again.. altough, in this case it does most probably only invoke `reader#setValidating`..

Reply:
If we take ClassPathXmlApplicationContext as a blackbox I would certainly wrap the resolver into our own implementation before passing it into super. Therefore the solution to call super at the end was explicitly chosen and actually I would want to stick with it.